### PR TITLE
Allow index null values

### PR DESCRIPTION
**Pull request by melissachang**
In index `condition_occurrence` table, some `end_date`s are null:

image

Our backend doesn't handle null, so we show an error on dataset preview:

image

In the past when this happened, I wrote sql to put `0` in index instead of `null`: #328 

However, it's better to change backend/frontend to support `null`. `Null` in index is common. Also, `0` means something different from `null`.

In this PR, I changed backend to support and propagate `null`s to frontend. Frontend displays `NULL`:

image

Shimon suggested these colors, which I thought were too distracting, so I went with former:

image

Also reverted #328 and manually confirmed UI still works.

Nice job with UI integration test, it found a bug in my code. :)

### CODE REVIEW

File: .github/workflows/ui-integration-test.yaml [hunk 1]
```diff
@@ -24,3 +24,18 @@ jobs:
           start: npm run fake
           wait-on: "http://localhost:3000"
           working-directory: ui
+      # NOTE: screenshots will be generated only if E2E test failed
+      # thus we store screenshots only on failures
+      # Alternative: create and commit an empty cypress/screenshots folder
+      # to always have something to upload
+      - uses: actions/upload-artifact@v2
```
[the remark this answers is not in the document]
Comment (melissachang) [reply .github/workflows/ui-integration-test.yaml:31]: Uploads screenshot and video to GHA page, so someone can download them.

image

**Review by melissachang — COMMENTED** [review]
Thanks

File: ui/src/components/treegrid.tsx [hunk 2]
```diff
@@ -317,9 +324,14 @@ function renderChildren(
                     paddingLeft: `${indent + 0.2}em`,
                   }),
                 }}
+                sx={{
+                  ...(value === "undefined" && {
```
[the remark this answers is not in the document]
Comment (melissachang) [reply ui/src/components/treegrid.tsx]: Changed to
```
                  ...(value === "NULL" && {
```

[the remark this answers is not in the document]
Comment (tjennison-google) [reply ui/src/components/treegrid.tsx]: Is there a reason you can't use `isNull` and avoid the string comparison which is dependent on the formatting of the output?

File: .github/workflows/ui-integration-test.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tjennison-google) [reply .github/workflows/ui-integration-test.yaml:31]: Neat, this is very helpful for debugging.

File: ui/src/components/treegrid.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (melissachang) [reply ui/src/components/treegrid.tsx]: Oh sorry, I didn't realize there was a variable called `isNull`. Done.

**Review by tjennison-google — APPROVED** [review]

**Pull request merged**